src/neuroconv/datainterfaces/behavior/video/externalvideointerface.py

```diff
@@ -0,0 +1,330 @@
+            a process which can take some time to complete.
+
+            To limit that scan to a small number of frames, set `stub_test=True`.
+        """
+        if self._timestamps is not None:
+            aligned_timestamps = [
+                timestamps + aligned_starting_time for timestamps in self.get_timestamps(stub_test=stub_test)
+            ]
+            self.set_aligned_timestamps(aligned_timestamps=aligned_timestamps)
+        elif self._segment_starting_times is not None:
+            self._segment_starting_times = [
+                segment_starting_time + aligned_starting_time for segment_starting_time in self._segment_starting_times
+            ]
+        else:
+            raise ValueError("There are no timestamps or starting times set to shift by a common value!")
+
+    def set_aligned_segment_starting_times(self, aligned_segment_starting_times: list[float], stub_test: bool = False):
+        """
+        Align the individual starting time for each video (segment) in this interface relative to the common session start time.
+
+        Must be in units seconds relative to the common 'session_start_time'.
+
+        Parameters
+        ----------
+        aligned_segment_starting_times : list of floats
+            The relative starting times of each video.
+        stub_test : bool, default: False
+            If timestamps have not been set to this interface, it will attempt to retrieve them
+            using the `.get_original_timestamps` method, which scans through each video;
+            a process which can take some time to complete.
+
+            To limit that scan to a small number of frames, set `stub_test=True`.
+        """
+        aligned_segment_starting_times_length = len(aligned_segment_starting_times)
+        assert aligned_segment_starting_times_length == self._number_of_files, (
+            f"The length of the 'aligned_segment_starting_times' list ({aligned_segment_starting_times_length}) does not match the "
+            "number of video files ({self._number_of_files})!"
+        )
+        if self._timestamps is not None:
+            self.set_aligned_timestamps(
+                aligned_timestamps=[
+                    timestamps + segment_starting_time
+                    for timestamps, segment_starting_time in zip(
+                        self.get_timestamps(stub_test=stub_test), aligned_segment_starting_times
+                    )
+                ]
+            )
+        else:
+            self._segment_starting_times = aligned_segment_starting_times
+
+    def align_by_interpolation(self, unaligned_timestamps: np.ndarray, aligned_timestamps: np.ndarray):
+        raise NotImplementedError("The `align_by_interpolation` method has not been developed for this interface yet.")
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        starting_frames: Optional[list[int]] = None,
+        module_name: Optional[str] = None,
```

Comment:
What about the "parent container" name for this that you proposed for other pipelines? Does it fit here?

Reply:
Only problem with `parent_container` is it constrains the options for processing module, which breaks back-compatibility with videointerface

Reply:
Can you elaborate?

Reply:
Previously, a user could specify and module_name (ex. 'my_arbitrarily_named_module'), but with parent_container we only have 'acquisition' or 'processing/behavior'.

Reply:
Is that a limitation of the other streams?
If so, I would prefer to unify and then expand. 

But if you want to keep it as it is currently I am fine with delaying this and address this separately.

Reply:
> Is that a limitation of the other streams as well (ephys and ophys)?

It is in #587 at least.

> If so, I would prefer to unify and then expand.
> But if you want to keep it as it is currently I am fine with delaying this and address this separately.

I would prefer to keep it limited (we don't really recommend making arbitrary processing modules anyway), but it depends on how important true back-compatibility is for this PR.

Reply:
I agree with your reasoning and we are on the same page here. We can generalize later if we need to but we a global solution instead.